shell/models/service.js

```diff
@@ -138,6 +138,11 @@ export default class Service extends SteveModel {
   }
 
   async fetchPods() {
+    if (!this.podRelationship) {
+      // If empty or not present, the service is assumed to have an external process managing its endpoints
```

Comment:
Is this assumption documented anywhere? I tried to look it up and I found some mentions of it but didn't see anything official.

Reply:
I grabbed it from the reference docs https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#servicespec-v1-core. It's not well indexed by google though


selector object
Route service traffic to pods with label keys and values matching this selector. If empty or not present, the service is assumed to have an external process managing its endpoints, which Kubernetes will not modify. Only applies to types ClusterIP, NodePort, and LoadBalancer. Ignored if type is ExternalName. More info: https://kubernetes.io/docs/concepts/services-networking/service/